src/util/android-interop.ts

```diff
@@ -0,0 +1,73 @@
+import { useCallback, useMemo } from 'react'
+
+import { storage, useStorage } from './storage'
+import { Ruleset } from '../game/rules/types'
+import { getSearchString } from '../game/url'
+import { logger } from './logger'
+
+export enum LocalStorage {
+    BehaviorSetting = 'android-interop-behavior',
+    AutoOpenApp = 'android-interop-auto-open-app',
+}
+
+export enum AndroidDetectBehavior {
+    AutoDetect = 'auto-detect',
+    ActAsAndroid = 'force-android',
+    ActAsNonAndroid = 'force-non-android'
+}
+
+
+export const androidDetectBehaviors = [
+    AndroidDetectBehavior.AutoDetect,
+    AndroidDetectBehavior.ActAsAndroid,
+    AndroidDetectBehavior.ActAsNonAndroid
+] as const
+
+export const isAndroidClient = (): boolean => {
+    const behavior = storage.getWithDefault({ key: LocalStorage.BehaviorSetting, defaultValue: AndroidDetectBehavior.AutoDetect, parser: v => v })
+
+    if (behavior === AndroidDetectBehavior.ActAsAndroid) return true
+    return behavior === AndroidDetectBehavior.AutoDetect && /Android/i.test(navigator.userAgent)
```

Comment:
If anyone knows of a better way of detecting Android and can post it, it would be most appreciated.